tls/s2n_x509_validator.c

```diff
@@ -376,6 +380,33 @@ s2n_cert_validation_code s2n_x509_validator_validate_cert_chain(struct s2n_x509_
 
         S2N_ERROR_IF(op_code <= 0, S2N_ERR_CERT_UNTRUSTED);
         validator->state = VALIDATED;
+        /* X509_STORE_CTX_get1_chain() returns a validated cert chain if a previous call to X509_verify_cert() was successful.
+         * X509_STORE_CTX_get0_chain() is a better API because it doesn't return a copy. But it's not available for Openssl 1.0.2.
```

Comment:
Could you call X509_STORE_CTX_get0_chain() to avoid the copy, but #define it to X509_STORE_CTX_get1_chain() for older OpenSSL? What happens if you do the cleanup even though X509_STORE_CTX_get0_chain() was used?

Reply:
After `X509_STORE_CTX_get1_chain`, the caller owns a reference count and has to call `_free` and with  `X509_STORE_CTX_get0_chain ` returns a pointer to some data structure without incrementing reference counters and caller does not call `_free`.